src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicIPConfig.cs

```diff
@@ -0,0 +1,291 @@
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.RecoveryServices.SiteRecovery.Properties;
+using Microsoft.Azure.Management.RecoveryServices.SiteRecovery.Models;
+
+namespace Microsoft.Azure.Commands.RecoveryServices.SiteRecovery
+{
+    /// <summary>
+    ///     Creates Azure Site Recovery VM NIC IP configuration for A2A replication.
+    /// </summary>
+    [Cmdlet("New", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "RecoveryServicesAsrVMNicIPConfig", DefaultParameterSetName = ASRParameterSets.AzureToAzure, SupportsShouldProcess = true)]
+    [Alias("New-ASRVMNicIPConfig")]
+    [OutputType(typeof(IPConfigInputDetails))]
+    public class NewAzureRmAsrVmNicIPConfig : SiteRecoveryCmdletBase
+    {
+        #region Parameters
+        /// <summary>
+        ///    Gets or sets the NIC Id.
+        /// </summary>
+        [Parameter(ParameterSetName = ASRParameterSets.AzureToAzure,
+            Mandatory = true,
+            HelpMessage = "Specify the ASR NIC GUID.")]
+        public string NicId { get; set; }
+
+        /// <summary>
+        ///    Specify the ASR Replication Protected Item.
+        /// </summary>
+        [Parameter(ParameterSetName = ASRParameterSets.AzureToAzure,
+            Mandatory = true,
+            HelpMessage = "Specify the ASR Replication Protected Item.")]
+        public ASRReplicationProtectedItem ReplicationProtectedItem { get; set; }
+
+
+        /// <summary>
+        ///    Gets or sets the IP config name.
+        /// </summary>
+        [Parameter(ParameterSetName = ASRParameterSets.AzureToAzure,
+            Mandatory = true,
+            HelpMessage = "Specify the IP config name.")]
+        public string IpConfigName { get; set; }
+
+        /// <summary>
+        ///    Gets or sets the if the IP config is primary on NIC.
+        /// </summary>
+        [Parameter(ParameterSetName = ASRParameterSets.AzureToAzure,
+            Mandatory = false,
+            HelpMessage = "Specify if the IP config is primary.")]
+        public bool IsPrimary { get; set; }
+
+        /// <summary>
+        ///     Gets or sets whether an existing IP config is selected for tfo/failover.
+        /// </summary>
+        [Parameter(ParameterSetName = ASRParameterSets.AzureToAzure,
+            Mandatory = false,
+            HelpMessage = "Specifies whether an existing IP config is selected for tfo/failover.")]
+        [ValidateNotNullOrEmpty]
```

Comment:
unnecessary

Reply:
but this is the param based on which ipconfig is selected/not-selected for failover.. or u mean validatenotnullorempty is not required !?

Reply:
the Validate attribute is not required

Reply:
done